group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -65,191 +94,233 @@ public String name() {
     }
 
     /**
-     * Pair of memberId and remaining partitions to meet the quota.
+     * Metadata for a topic including partition and subscription details.
      */
-    private static class MemberWithRemainingAssignments {
+    private static class TopicMetadata {
+        public final Uuid topicId;
+        public final int numPartitions;
+        public int numMembers;
+
+        public int minQuota = -1;
+        public int extraPartitions = -1;
+        public int nextRange = 0;
+
+        /**
+         * Constructs a new TopicMetadata instance.
+         *
+         * @param topicId           The topic Id.
+         * @param numPartitions     The number of partitions.
+         * @param numMembers        The number of subscribed members.
+         */
+        private TopicMetadata(Uuid topicId, int numPartitions, int numMembers) {
+            this.topicId = topicId;
+            this.numPartitions = numPartitions;
+            this.numMembers = numMembers;
+        }
+
         /**
-         * Member Id.
+         * Factory method to create a TopicMetadata instance.
+         *
+         * @param topicId           The topic Id.
+         * @param numPartitions     The number of partitions.
+         * @param numMembers        The number of subscribed members.
+         * @return A new TopicMetadata instance.
          */
-        private final String memberId;
+        public static TopicMetadata create(Uuid topicId, int numPartitions, int numMembers) {
```

Comment:
Does it provide anything more that calling the constructor directly? It seems that they have the same params.